src/matlab2tikz.m

```diff
@@ -1677,16 +1674,18 @@ function assertRegularAxes(handle)
     % Check if any value is infinite/NaN. In that case, add appropriate option.
     m2t = jumpAtUnboundCoords(m2t, data);
 
-    [m2t, str] = writePlotData(m2t, str, data, drawOptions);
-    [m2t, str] = addLabel(m2t, str, h);
+    [m2t, dataString]  = writePlotData(m2t, data, drawOptions);
+    [m2t, labelString] = addLabel(m2t, h);
+    
+    str = [dataString, labelString];
 end
 % ==============================================================================
-function [m2t, str] = specialDrawZplaneUnitCircle(m2t, h, drawOptions)
+function [m2t, str] = specialDrawZplaneUnitCircle(m2t, drawOptions)
     % Draw unit circle and axes.
 
     % TODO Don't hardcode "10", but extract from parent axes of |h|
```

Comment:
In the light of this comment, I would keep passing in this `h` into the function. If you want, you can add an `%#ok` after the function declaration to suppress the message (it will be re-enabled automatically once we do use `h`)